### DISCUSSION TITLE
Suggest to replace maxminddb gem with maxmind-geoip2

### DISCUSSION THREAD

**Discussion by anthonypwinning** [alexreisner/geocoder#1538] [Ideas]
Currently _geocoder_ requires the [_maxminddb_ gem](https://github.com/yhirose/maxminddb) which does not look like it has had an [update in several years](https://rubygems.org/gems/maxminddb/). I suggest using the official [_maxmind-geoip2_](https://github.com/maxmind/GeoIP2-ruby) which is [getting regular updates](https://rubygems.org/gems/maxmind-geoip2).

**Comment by alexreisner** [comment]
Yes, thanks for pointing this out. I agree. Open to pull requests.

**Reply by anthonypwinning** [reply]
Excellent. I'll give it a stonking red hot go. Strap yourself in.

**Reply by alexreisner** [reply]
Strapped, belted, and braced. Do your worst.

**Reply by anthonypwinning** [reply]
Argh... This turned out to be trickier than I thought. It would appear that the _maxminddb_ gem and the _hive_geoip2_ gem had very similar methods and syntax. This isn't the case with the _maxmind-geoip2_, meaning that either I have to mangle things together, and end up with an inelegant solution, or split support for _maxmind-geoip2_ into its own separate thing (which seems a bit silly). I'm going to need to have a think a breather...

**Comment by alexreisner** [comment]
No worries, take your time. I haven't looked at this in detail but it's possible it's too complex to be worth doing at the moment.
